### PR TITLE
Add support for openstack-must-gather container building

### PR DESCRIPTION
**Pull request by lewisdenny**
This patch adds the ability for the openstack-must-gather container
to be build for PRs that contain a openstack-must-gather patch.

This means any PR created in openstack-must-gather or depending on a
change from openstack-must-gather repo will have a freshly build
container at the head of the patch.

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: ci_framework/roles/os_must_gather/tasks/build_openstack-must-gather_image.yml
Comment (lewisdenny) [reply ci_framework/roles/os_must_gather/tasks/build_openstack-must-gather_image.yml:33]: done

File: ci_framework/roles/os_must_gather/tasks/build_openstack-must-gather_image.yml
Comment (raukadah) [inline ci_framework/roles/os_must_gather/tasks/build_openstack-must-gather_image.yml:25]:
```suggestion
    chdir: {{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/openstack-must-gather"
```

File: ci_framework/roles/os_must_gather/tasks/build_openstack-must-gather_image.yml
Comment (viroel) [inline ci_framework/roles/os_must_gather/tasks/build_openstack-must-gather_image.yml:30]: this should be a role variable, so we can replace with a registry from content provider, for instance.

File: ci_framework/roles/os_must_gather/tasks/main.yml
Comment (viroel) [inline ci_framework/roles/os_must_gather/tasks/main.yml:31]: Maybe, if we use a local registry, I think that this variable is not set. It is better to control when or not to push with a control var, inside the playbook?

File: ci_framework/roles/os_must_gather/tasks/main.yml
Comment (viroel) [inline ci_framework/roles/os_must_gather/tasks/main.yml:30]: why not?

**Review by viroel — COMMENTED** [review]
Thanks Lewis, some comments inline

File: ci_framework/roles/os_must_gather/tasks/build_openstack-must-gather_image.yml
Comment (raukadah) [inline ci_framework/roles/os_must_gather/tasks/build_openstack-must-gather_image.yml:21]: We can also add one more ansible var to store the must-gather repo path. It will be useful for local development having different local source path.

**Comment by lewisdenny** [comment]
recheck

File: ci_framework/roles/os_must_gather/tasks/main.yml
Comment (lewisdenny) [reply ci_framework/roles/os_must_gather/tasks/main.yml]: I have it commented out just to test CI, without it commented the image wouldn't build because openstack-must-gather wouldn't be in the `zuul_change_list` but maybe you are indicating I could use a depends-on here?

**Comment by lewisdenny** [comment]
Thank you for the review @viroel @raukadah !

**Comment by lewisdenny** [comment]
recheck

File: ci_framework/roles/os_must_gather/tasks/main.yml
Comment (raukadah) [inline ci_framework/roles/os_must_gather/tasks/main.yml:37]: It is no longer needed, if we use the zuul provided repo path. "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/openstack-must-gather". May be we can replace this zuul piece with a new var to `cifmw_os_must_gather_image_build`.

Comment (raukadah) [reply ci_framework/roles/os_must_gather/tasks/main.yml:37]: Sorry for the noise here. I withdraw my comment. If we always builds the image and push it to registry. Then it is going to be a waste of resources. It make sense to build it only on zuul changes.

**Comment by lewisdenny** [comment]
Can I get a lgtm if you happy @raukadah ?

**Review by chkumar246 — APPROVED** [review]
/approve

**Review by raukadah — APPROVED** [review]
/approve

**Pull request merged**
